### PR TITLE
LibGfx/PNG: Read metadata from the eXIf chunk

### PR DESCRIPTION
**Pull request by LucasChollet**

### CODE REVIEW

File: Tests/LibGfx/TestImageDecoder.cpp [hunk 1]
```diff
@@ -768,6 +769,18 @@ TEST_CASE(test_png)
     TRY_OR_FAIL(expect_single_frame(*plugin_decoder));
 }
 
+TEST_CASE(test_exif)
+{
+    auto file = TRY_OR_FAIL(Core::MappedFile::map(TEST_INPUT("png/exif.png"sv)));
```
Comment (nico) [inline Tests/LibGfx/TestImageDecoder.cpp:774]: You might've forgotten to `git add` this file. (Please mention how you created it somewhere.)

Comment (LucasChollet) [reply Tests/LibGfx/TestImageDecoder.cpp:774]: Oops, sorry about that!

**Review by nico — APPROVED** [review]
lgtm!

I think this means that we now fail to load pngs with invalid exif data. If that happens in practice, we'll probably want to log something about the invalid exif data and still decode the image data. (Per https://www.w3.org/TR/2003/REC-PNG-20031110/#5Chunk-naming-conventions , we probably want to log-and-ignore errors in all chunks that start with a lower-case letter.)

**Comment by LucasChollet** [comment]
I never searched for the reason behind the different cases in the names, thanks for the link!
The error management is noted for future me.

File: Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp [hunk 2]
```diff
@@ -1235,6 +1245,8 @@ static ErrorOr<void> process_chunk(Streamer& streamer, PNGLoadingContext& contex
         return process_fcTL(chunk_data, context);
     if (chunk_type == "fdAT"sv)
         return process_fdAT(chunk_data, context);
+    if (chunk_type == "eXIf"sv)
+        return process_eXif(chunk_data, context);
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp:1249]: Not sure if this matters but the case is different between eXIf and eXif.

Comment (LucasChollet) [reply Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp]: > Not sure if this matters
In practice I guess no, but they "eXif" and "eXIf" are technically different chunks.
Thanks for catching that!

**Pull request merged**